### ISSUE TITLE
Async lambda function error: space-before-function-paren

### ISSUE THREAD

**Issue by marianspoiala** [eslint/eslint#10769]
**Tell us about your environment**

* **ESLint Version:5.3.0**
* **Node Version:8.11.3**
* **npm Version:5.6.0**

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
       "parserOptions": {
               "ecmaVersion": 2018
       },
       "rules": {
               "space-before-function-paren": ["error", "never"]
       }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
function hello() {
   return 0;
}

(async () => {
   await hello();
})();
```

```bash
node_modules/.bin/eslint index.js
```

**What did you expect to happen?**
No errors should appear.

**What actually happened? Please include the actual, raw output from ESLint.**
I got the following error: "Unexpected space before function parentheses  space-before-function-paren"

**Comment by platinumazure** [comment]
Hi @marianspoiala, thanks for the issue.

You didn't specify on which line the error occurs, but my guess is the problem is on the async function opening line.

Try something like this:

```js
function hello() {
   return 0;
}

(async() => {
   await hello();
})();
```

Or, if you want `async ()` to be allowed but otherwise forbid spaces before function parentheses, you should configure the rule like this:

```json
{
    "rules": {
        "space-before-function-paren": ["error", {
            "anonymous": "never",
            "named": "never",
            "asyncArrow": "ignore"
        }]
    }
}
```

(Or use "always" to require `async ()` and forbid `async()`.)
